### PR TITLE
Devb

### PR DESCRIPTION
**Pull request by benroywillis**

### CODE REVIEW

File: Python/KernelDetector/EncodeDetect_new.py [hunk 1]
```diff
@@ -59,9 +59,11 @@ def DetectKernels(sourceFile, thresh = 0.95, hotThresh = 512, newLine = False):
         for sub in blockMap[key]:
```
Comment (ruhrie) [inline Python/KernelDetector/EncodeDetect_new.py]: This entire project should be deleted now

File: cartographer/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+file(GLOB SOURCES "*.cpp")
+add_executable(cartographer ${SOURCES})
+
+set_target_properties(cartographer
+	PROPERTIES
+    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin" 
+)
+target_link_libraries(cartographer PRIVATE ${ZLIB_LIBRARY} ${llvm_libs} nlohmann_json nlohmann_json::nlohmann_json)
```
Comment (ruhrie) [inline cartographer/CMakeLists.txt]: Does cartographer link against LLVM?

File: cartographer/EncodeDetect.h [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (ruhrie) [inline cartographer/EncodeDetect.h]: A few things on this file:
1. add `#pragma once` on top
2. the structs on 8/9 should be defined as external and actually defined in the cpp
3. Why is there iostream here?

File: cartographer/EncodeExtract.cpp [hunk 4]
```diff
@@ -0,0 +1,205 @@
+
```
Comment (ruhrie) [inline cartographer/EncodeExtract.cpp]: This file hasn't been formatted yet

File: cartographer/EncodeExtract.h [hunk 5]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (ruhrie) [inline cartographer/EncodeExtract.h]: `#pragma once`
Remove set

File: cartographer/cartographer.cpp [hunk 6]
```diff
@@ -0,0 +1,56 @@
+
```
Comment (ruhrie) [inline cartographer/cartographer.cpp]: Remove new line at top

File: cartographer/cartographer.cpp [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    for( auto entry : type2Kernels )
+    {
+        for( auto index : entry.second )
+        {
+            std::cout << index << " , ";
+        }
+        std::cout << "\n\n";
+    }
+    */
```
Comment (ruhrie) [inline cartographer/cartographer.cpp]: Remove comment block

File: tik/Util.cpp [hunk 8]
```diff
@@ -14,6 +15,15 @@ void PrintVal(llvm::Value *val)
     std::cout << str << "\n";
 }
 
+void PrintVal(llvm::Module *mod)
+{
+    AssemblyAnnotationWriter *write = new llvm::AssemblyAnnotationWriter();
+    std::string str;
+    llvm::raw_string_ostream rso(str);
+    mod->print(rso, write);
+    std::cout << str;
```
Comment (ruhrie) [inline tik/Util.cpp]: I think this needs a new line at the end.

File: tik/include/tik/Util.h [hunk 9]
```diff
@@ -4,13 +4,15 @@
 #include <llvm/IR/BasicBlock.h>
```
Comment (ruhrie) [inline tik/include/tik/Util.h]: Please add `#pragma once`

File: cartographer/CMakeLists.txt (hunk 2 above)
Comment (benroywillis) [reply cartographer/CMakeLists.txt]: Uses their cl:: command parser

File: Python/KernelDetector/EncodeDetect_new.py (hunk 1 above)
Comment (benroywillis) [reply Python/KernelDetector/EncodeDetect_new.py]: Done

File: cartographer/EncodeExtract.cpp (hunk 4 above)
Comment (benroywillis) [reply cartographer/EncodeExtract.cpp]: Done

File: cartographer/cartographer.cpp (hunk 6 above)
Comment (benroywillis) [reply cartographer/cartographer.cpp]: Done for all files

File: cartographer/EncodeDetect.h (hunk 3 above)
Comment (benroywillis) [reply cartographer/EncodeDetect.h]: 1.) done
2.) not needed
3.) Removed

File: cartographer/cartographer.cpp (hunk 7 above)
Comment (benroywillis) [reply cartographer/cartographer.cpp]: done

File: tik/Util.cpp (hunk 8 above)
Comment (benroywillis) [reply tik/Util.cpp]: done

File: tik/include/tik/Util.h (hunk 9 above)
Comment (benroywillis) [reply tik/include/tik/Util.h]: at the top of this header already

File: .gitignore [hunk 10]
```diff
@@ -1,4 +1,5 @@
 build/
+build2/
```
Comment (ruhrie) [inline .gitignore]: Why is this here
